tests/publiccloud/ipa.pm

```diff
@@ -159,6 +138,80 @@ sub post_fail_hook {
     assert_script_run("aws ec2 delete-key-pair --key-name " . $self->{'ipa_ssh_key_name'});
 }
 
+sub az_run_ipa {
```

Comment:
just thinking out loud, don't consider this as some action time. 
If it would be some real object-oriented language like Java / C# I would recommend you here to create parent class with set of abstract methods (`run_ipa`, `cleanup` etc. )  and two child classes (az,ec2 ) which implement abstract methods.  But sad story that our reality is perl :( so I would dare even to suggest something like. Also have no idea how to implement this in this freaky language

Reply:
Actually I'm thinking about some kind of abstraction. But as long as I use ipa only in one file, I do not find it helpful.